itou/openid_connect/models.py

```diff
@@ -101,11 +101,20 @@ def check_valid_kind(self, user, user_data_dict, is_login):
         if user.kind not in self.login_allowed_user_kinds or (user.kind != user_data_dict["kind"] and not is_login):
             raise InvalidKindException(user)
 
+    def handle_email_overloaded_on_new_identity(self, user):
+        """
+        If there is already a user with this username (user_info_dict["sub"])
+        and from the same identity provider, by default we update and return it.
+
+        Some providers allow for emails to be re-used on multiple accounts,
+        and this hook can be used to prevent the default behaviour by raising
+        an exception.
+        """
+        return
+
     def create_or_update_user(self, is_login=False):
         """
         Create or update a user managed by another identity provider.
-         - If there is already a user with this username (user_info_dict["sub"])
-           and from the same identity provider, we update and return it.
          - If there is already a user with the email, we return this user.
```

Comment:
Ici on peut ajouter `except if it's from the same identity provider but with a different username (as FranceTravail Connect allows multiple accounts with the same email)